cmd/create/cluster/cmd.go

```diff
@@ -764,105 +766,71 @@ func run(cmd *cobra.Command, _ []string) {
 	hasRoles := false
 	if isSTS && roleARN == "" {
 		minor := ocm.GetVersionMinor(version)
-		role := aws.AccountRoles[aws.InstallerAccountRole]
 
+		var spin *spinner.Spinner
+		if r.Reporter.IsTerminal() {
+			spin = spinner.New(spinner.CharSets[9], 100*time.Millisecond)
+		}
+		if spin != nil {
+			if !output.HasFlag() && r.Reporter.IsTerminal() {
+				r.Reporter.Infof("Fetching account roles for the aws account: %s", awsCreator.AccountID)
+			}
+			spin.Start()
+		}
 		// Find all installer roles in the current account using AWS resource tags
-		roleARNs, err := awsClient.FindRoleARNs(aws.InstallerAccountRole, minor)
+		currentArnBundles, err := awsClient.FindAccountRoleArnBundles(minor)
 		if err != nil {
-			r.Reporter.Errorf("Failed to find %s role: %s", role.Name, err)
+			r.Reporter.Errorf("Failed to find account role bundles: %s", err)
 			os.Exit(1)
 		}
+		if spin != nil {
+			spin.Stop()
+		}
 
-		if len(roleARNs) > 1 {
-			defaultRoleARN := roleARNs[0]
-			// Prioritize roles with the default prefix
-			for _, rARN := range roleARNs {
-				if strings.Contains(rARN, fmt.Sprintf("%s-%s-Role", aws.DefaultPrefix, role.Name)) {
-					defaultRoleARN = rARN
-				}
-			}
-			r.Reporter.Warnf("More than one %s role found", role.Name)
-			if !interactive.Enabled() && confirm.Yes() {
-				r.Reporter.Infof("Using %s for the %s role", defaultRoleARN, role.Name)
-				roleARN = defaultRoleARN
-			} else {
-				roleARN, err = interactive.GetOption(interactive.Input{
-					Question: fmt.Sprintf("%s role ARN", role.Name),
-					Help:     cmd.Flags().Lookup(role.Flag).Usage,
-					Options:  roleARNs,
-					Default:  defaultRoleARN,
-					Required: true,
-				})
-				if err != nil {
-					r.Reporter.Errorf("Expected a valid role ARN: %s", err)
-					os.Exit(1)
-				}
-			}
-		} else if len(roleARNs) == 1 {
-			if !output.HasFlag() || r.Reporter.IsTerminal() {
-				r.Reporter.Infof("Using %s for the %s role", roleARNs[0], role.Name)
-			}
-			roleARN = roleARNs[0]
-		} else {
+		if len(currentArnBundles) == 0 {
 			r.Reporter.Warnf("No account roles found. You will need to manually set them in the " +
 				"next steps or run 'rosa create account-roles' to create them first.")
 			interactive.Enable()
-		}
-
-		if roleARN != "" {
-			// Get role prefix
-			rolePrefix, err := getAccountRolePrefix(roleARN, role)
-			if err != nil {
-				r.Reporter.Errorf("Failed to find prefix from %s account role", role.Name)
-				os.Exit(1)
+		} else {
+			possibleBundles := make([]string, 0, 1)
```

Comment:
Why do we need the `possibleBundles` slice?
Can we sort the `currentArnBundles` and choose the first one to prompt to the user?

Reply:
The possible bundles only show the prefix key so the user can take a look at the possibilities when choosing from the interactive menu